### PR TITLE
Use merged measure units

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-776

### What?

I have added/removed/altered:

- [x] Make sure we fetch the merged measure units when deciding which ones apply in the delta case
- [x] Move methods that are shared/apply to the commodity to the commodity helper
- [x] Adds specs for the moved methods

### Why?

I am doing this because:

- This is currently broken when the source commodity doesn't have applicable measure units but the other commodity does

### CODE REVIEW

File: spec/helpers/commodity_helper_spec.rb [hunk 1]
```diff
@@ -188,4 +188,68 @@
...
+          'measurement_unit_code' => 'DTN',
+          'measurement_unit_qualifier_code' => '',
+          'unit' => 'x 100 kg',
+          'unit_hint' => 'Enter the value in decitonnes (100kg)',
+          'unit_question' => 'What is the weight of the goods you will be importing?',
+        },
+      }
+    end
+
+    context 'when on deltas are applicable' do
```
Comment (willfish) [inline spec/helpers/commodity_helper_spec.rb:207]:
```suggestion
    context 'when deltas are applicable' do
```

File: spec/helpers/commodity_helper_spec.rb [hunk 2]
```diff
@@ -188,4 +188,68 @@
...
+
+        expect(Api::Commodity).to have_received(:build).with('xi', anything, anything)
+      end
+
+      it 'fetches the measure units' do
+        expect(helper.applicable_measure_units).to eq(expected_measure_units)
+      end
+    end
+
+    context 'when on deltas are not applicable' do
```
Comment (willfish) [inline spec/helpers/commodity_helper_spec.rb:227]:
```suggestion
    context 'when deltas are not applicable' do
```

File: spec/helpers/commodity_helper_spec.rb [hunk 3]
```diff
@@ -188,4 +188,68 @@
...
+      }
+    end
+
+    context 'when deltas are applicable' do
+      let(:user_session) { build(:user_session, :with_commodity_information, :deltas_applicable) }
+
+      it 'fetches the uk commodity' do
+        helper.applicable_measure_units
+
+        expect(Api::Commodity).to have_received(:build).with('uk', anything, anything)
```
Comment (tavy88) [inline spec/helpers/commodity_helper_spec.rb:213]: Is `anything` this a reserved keyword?

Comment (willfish) [reply spec/helpers/commodity_helper_spec.rb:213]: It instantiates a matcher of the type RSpec::Mocks::ArgumentMatchers::AnyArgMatcher. Basically it's a method call that returns a AnyArgMatcher which makes any argument of any type pass comparison

**Review by tavy88 — APPROVED** [review]

**Pull request merged**